### PR TITLE
[CDAP-19479] Add editing stage column to deployed pipeline list

### PR DESCRIPTION
**Pull request by sumengwang**
# [CDAP-19479] Add editing stage column to pipeline draft list

## Description
* When there's no draft saved
image
* When there's a draft
image
* When draft is outdated
image
* When there's a draft and the user click edit
image

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-19479](https://cdap.atlassian.net/browse/CDAP-19479)

[CDAP-19479]: https://cdap.atlassian.net/browse/CDAP-19479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sanjanasandeep — APPROVED** [review]

File: app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsDetailsActions/index.js [hunk 1]
```diff
@@ -39,7 +42,7 @@ const PipelineDetailsDetailsActions = ({
   config,
   version,
 }) => {
-  const lifecycleManagementEditEnabled = useFeatureFlagDefaultFalse(
+  const lifecycleManagementEditEnabled = useFeatureFlagDefaultTrue(
```
Comment (de-lan) [inline app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsDetailsActions/index.js]: So we turn on the feature flag by default now?

**Review by de-lan — APPROVED** [review]

File: app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/PipelineTableRow.tsx [hunk 2]
```diff
@@ -24,6 +24,9 @@ import LastStart from 'components/PipelineList/DeployedPipelineView/LastStart';
 import RunsCount from 'components/PipelineList/DeployedPipelineView/RunsCount';
 import DeployedActions from 'components/PipelineList/DeployedPipelineView/DeployedActions';
 import { IPipeline } from 'components/PipelineList/DeployedPipelineView/types';
+import { useSelector } from 'react-redux';
+import { MyPipelineApi } from 'api/pipeline';
+import { Observable } from 'rxjs/Observable';
```
Comment (njbriggs) [inline app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/PipelineTableRow.tsx]: Are these last 2 imports used?

File: app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsDetailsActions/PipelineDetailsActionsButton/index.js [hunk 3]
```diff
@@ -98,6 +102,40 @@ export default class PipelineDetailsActionsButton extends Component {
...
+  };
+
+  discardAndStartNewEdit = () => {
+    this.toggleDiscardConfirmation();
+    editPipeline(this.props.pipelineName);
+  };
+
+  handlePipelineEdit = () => {
+    if (!this.props.editDraftId) {
+      this.editConfigAndNavigate(this.props.pipelineName);
```
Comment (de-lan) [inline app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsDetailsActions/PipelineDetailsActionsButton/index.js]: seems `editConfigAndNavigate` is only called in this func, then no need to have it on line139?

simple put `editPipeline(this.props.pipelineName, sanitizeConfig(this.pipelineConfig));`

File: app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/PipelineTableRow.tsx [hunk 4]
```diff
@@ -33,36 +34,63 @@ interface IProps {
...
+      // draft parentVersion is the same as the latest appVersion
+      setEditStageMessage('Editing in progress');
+    } else {
+      setEditStageMessage('Draft out of date');
+    }
+    setDraftId(filteredDraft[0].id);
+  }, [drafts]);
+
+  return (
+    <a href={pipelineLink} onClick={saveDraftIdToLocalStorage} className=" grid-row">
```
Comment (de-lan) [inline app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/PipelineTableRow.tsx]: extra space in `className`

File: app/cdap/components/PipelineList/DeployedPipelineView/store/ActionCreator.ts [hunk 5]
```diff
@@ -25,6 +25,7 @@ import { IPipeline } from 'components/PipelineList/DeployedPipelineView/types';
 import { objectQuery } from 'services/helpers';
 import { PROGRAM_STATUSES, GLOBALS } from 'services/global-constants';
 import debounce from 'lodash/debounce';
+import { IDraft } from 'components/PipelineList/DraftPipelineView/types';
```
Comment (de-lan) [inline app/cdap/components/PipelineList/DeployedPipelineView/store/ActionCreator.ts:23]: nit: there are bunch of unused imports

**Review by njbriggs — APPROVED** [review]

**Pull request merged**